src/google/protobuf/compiler/csharp/csharp_helpers.h

```diff
@@ -158,6 +158,34 @@ inline bool IsProto2(const FileDescriptor* descriptor) {
   return descriptor->syntax() == FileDescriptor::SYNTAX_PROTO2;
 }
 
+inline bool SupportsPresenceApi(const FieldDescriptor* descriptor) {
+  // Not quite the same as descriptor->is_singular_with_presence() in two ways:
+  
+  // In proto3 we add field presence only for explicitly-optional fields. Message fields
+  // don't get field presence by default, because the user can just compare with a null reference.
```

Comment:
If it's not too much trouble, it would be helpful to keep the CL (and the C# code generator generally) to 80 chars to match Google C++ style: https://google.github.io/styleguide/cppguide.html#Line_Length

Otherwise we'll have to reformat it later to pass google3 ClangTidy checks, etc.

Reply:
Sure, happy to. (That will probably be tomorrow though.)